test/model/autocomplete_checks.dart

```diff
@@ -3,14 +3,28 @@ import 'package:zulip/model/autocomplete.dart';
 import 'package:zulip/widgets/compose_box.dart';
 
 extension ComposeContentControllerChecks on Subject<ComposeContentController> {
-  Subject<AutocompleteIntent?> get autocompleteIntent => has((c) => c.autocompleteIntent(), 'autocompleteIntent');
+  Subject<AutocompleteIntent<MentionAutocompleteQuery>?> get autocompleteIntent => has((c) => c.autocompleteIntent(), 'autocompleteIntent');
 }
 
-extension AutocompleteIntentChecks on Subject<AutocompleteIntent> {
+extension ComposeTopicControllerChecks on Subject<ComposeTopicController> {
+  Subject<AutocompleteIntent<TopicAutocompleteQuery>?> get autocompleteIntent => has((c) => c.autocompleteIntent(), 'autocompleteIntent');
+}
+
+extension AutocompleteIntentChecks on Subject<AutocompleteIntent<MentionAutocompleteQuery>> {
```

Comment:
This doesn't look right — the extension is named as if it's on AutocompleteIntent in general, but it's only on a specific subtype.

```suggestion
extension AutocompleteIntentChecks on Subject<AutocompleteIntent> {
```

You'll have to update a getter to match the changes to the class.

Similarly in `test/widgets/compose_box_checks.dart`.